packages/block-editor/src/store/index.js

```diff
@@ -35,11 +35,7 @@ export const store = createReduxStore( STORE_NAME, {
 	persist: [ 'preferences' ],
 } );
 
-// We will be able to use the `register` function once we switch
-// the "preferences" persistence to use the new preferences package.
-const registeredStore = registerStore( STORE_NAME, {
-	...storeConfig,
-	persist: [ 'preferences' ],
-} );
-unlock( registeredStore ).registerPrivateActions( privateActions );
-unlock( registeredStore ).registerPrivateSelectors( privateSelectors );
+unlock( store ).registerPrivateActions( privateActions );
+unlock( store ).registerPrivateSelectors( privateSelectors );
+
+register( store );
```

Comment:
You need to keep the legacy `registerStore` here. Otherwise the `persist` plugin won't be active. It works by overriding the `registerStore` method of the `registry` and reading the `persist` option. If you call `register`, the plugin won't catch it.

This will be solved once we finish migration of the `block-editor` store to the `preferences` package. It's being worked on in #39632.

Reply:
Ah, bummer, I see.

I don't think we can remove this workaround yet then because if we do this:

```js
export const store = createReduxStore( STORE_NAME, {
	...storeConfig,
	persist: [ 'preferences' ],
} );

unlock( store ).registerPrivateActions( privateActions );
unlock( store ).registerPrivateSelectors( privateSelectors );

registerStore( STORE_NAME, {
	...storeConfig,
	persist: [ 'preferences' ],
} );
```

Then the private selectors are not callable from public selectors as they are not bound to the store that's created by `registerStore`.

(When I try it I get a `Uncaught TypeError: selector.registry is undefined` error on load.)

Reply:
I suppose we will wait until https://github.com/WordPress/gutenberg/pull/39632 is merged!

Reply:
I did start looking at https://github.com/WordPress/gutenberg/pull/39632 one again. I've rebased it and hope to push some further options this week.